public/static/js/spawned_hosts.js

```diff
@@ -166,12 +167,26 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope', '$window', '$timeout', '$q',
       $scope.updateSpawnedHosts();
     }, 60000);
 
-    // Returns true if the user can spawn another host. If hosts has not been initialized it
+    // Returns true if the user can spawn another host. If hosts have not been initialized it
     // assumes true.
     $scope.availableHosts = function () {
       return ($scope.hosts == null) || ($scope.hosts.length < $scope.maxHostsPerUser)
     }
 
+    $scope.availableUnexpirableHosts = function () {
+      var available = $scope.maxUnexpirableHostsPerUser;
+      for (host of $scope.hosts) {
```

Comment:
Just to verify: $scope.hosts data doesn't get dirty right? So if there's one unexpirable host and the user checkmarks the box it won't disappear because this function is now 0?

Reply:
I think the original_expiration field is unchanged.
If the user uses the last unexpirable host the function will return 1 until they click the update button.